common/Windows/WindowTitleBar.xaml

```diff
@@ -0,0 +1,67 @@
+<UserControl
```

Comment:
Hmm correct me if wrong but this should be more generic right?  Like how you did the secondary window template . I believe we should do it in a way where the WindowTitleBar accepts a user control only.  For example if someone wanted to put a searchbox in the center of the window like in vs code it's as simple as creating a `common/WindowTitleBarTemplates/WindowTitleBarWithSearchBox.xaml`  and doing:

```xaml
    <Windows:SecondaryWindow.WindowTitleBar>
        <WindowTitleBarTemplates:WindowTitleBarWithSearchBox
             ..property details go here
        />
    </Windows:SecondaryWindow.WindowTitleBar>
```

but if they just want the defaults that everywhere else will use they do:
```xaml
    <Windows:SecondaryWindow.WindowTitleBar>
        <WindowTitleBarTemplates:DefaultWindowTitleBar
             ..property details go here
        />
    </Windows:SecondaryWindow.WindowTitleBar>
```

But if I miss understood something and this is achievable as currently constructed let me know. By the looks of it if we'd want to add newer features to the title bar we'd have to keep editing the WindowTitleBar class directly. I could imagine a scenario where maybe for one window the height could change dynamically based on its content but for default se cases the height isn't going to change.

Reply:
This was actually in my original design ([previous commit](https://github.com/microsoft/devhome/pull/1589/commits/36db7b2357fc84d2b1bd955f593bc94b3ac5ebf8#diff-1a130b963fdad9168642d11a90c922d2e594a93593d028289b6d251e9cce9295R57-R73)) 😆 However, I changed it after noticing that it's currently challenging to have mouse interaction in the title bar as the entire area becomes a draggable zone for the window.

- https://github.com/microsoft/microsoft-ui-xaml/issues/4233